packages/zoe/src/contractSupport/bondingCurves.js

```diff
@@ -18,23 +19,26 @@ const BASIS_POINTS = 10000n; // TODO change to 10_000n once tooling copes.
  * request, and to do the actual reallocation after an offer has
  * been made.
  *
- * @param {bigint} inputValue - the value of the asset sent
+ * @param {Value} inputValue - the value of the asset sent
```

Comment:
Why does Typescript want this change?  Is it because we can't tell it that the Amounts we started with are from NatMath types?

Reply:
Yeah, Values in Amounts are either NatValues (bigints) or SetValues (arrays), so when we as programmers know that we are using a bigint, the type system doesn't know. We have two choices: do an `assert.typeof(value, `bigint`)` before we call the method that requires bigints, or we can loosen the type of the method to be a Value. That's what I chose for this method, otherwise I would have to do 3 or so asserts. We may be able to improve the types, but this is the best I know how to do. The plus side is that instead of Values being `any` and allowing anything in amounts, we will catch many more problems in linting.